### PR TITLE
docs: unwrap nested named union and literal types

### PR DESCRIPTION
**Pull request by ST-DDT**
Created based on the requested https://github.com/faker-js/faker/pull/1958#discussion_r1145578892

This has an impact on these methods:

```txt
Processing Module Color
- cmyk
- colorByCSSColorSpace
- hsl
- hwb
- lab
- lch
- rgb
Processing Module Helpers
- unique
Processing Module Internet
- emoji
- httpStatusCode
- url
Processing Module Random
- alpha
- alphaNumeric
- numeric
Processing Module String
- alpha
- alphanumeric
- hexadecimal
- numeric
```

by unwrapping these types:

````txt
AlphaChar -> 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z' | 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'
AlphaNumericChar -> '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z' | 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'
Casing -> 'lower' | 'mixed' | 'upper'
ColorFormat -> 'binary' | 'css' | 'decimal'
EmojiType -> 'activity' | 'body' | 'flag' | 'food' | 'nature' | 'object' | 'person' | 'smiley' | 'symbol' | 'travel'
HTTPProtocolType -> 'http' | 'https'
HTTPStatusCodeType -> 'clientError' | 'informational' | 'redirection' | 'serverError' | 'success'
LowerAlphaChar -> 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'
NumberColorFormat -> 'decimal'
NumericChar -> '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9'
RecordKey -> number | string | symbol
StringColorFormat -> 'binary' | 'css'
UpperAlphaChar -> 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z'
````

### CODE REVIEW

**Comment by ST-DDT** [comment]
IMO we should not merge this as is, especially the `Char` types.

grafik

**Comment by matthewmayer** [comment]
Perhaps there should be a ~~white list~~ allow list of types which are unwrapped or a ~~blacklist~~ bock/deny list of types which are not.

File: scripts/apidoc/signature.ts [hunk 1]
```diff
@@ -298,9 +298,18 @@ function typeToText(type_?: Type, short = false): string {
         .sort()
         .join(' | ');
 
-    case 'reference':
+    case 'reference': {
```
Comment (ST-DDT) [inline scripts/apidoc/signature.ts]:
```suggestion
    case 'reference':
```

File: scripts/apidoc/signature.ts [hunk 2]
```diff
@@ -311,6 +320,7 @@ function typeToText(type_?: Type, short = false): string {
           .map((t) => typeToText(t, short))
           .join(', ')}>`;
       }
+    }
```
Comment (ST-DDT) [inline scripts/apidoc/signature.ts]:
```suggestion
```

**Comment by ST-DDT** [comment]
**Team Decision**

We will block type expansion for all `*Char` types, but otherwise expand them.

**Comment by ST-DDT** [comment]
Docs-Diff:

````txt
Documentation changes detected:
- Color
  - cmyk
  - colorByCSSColorSpace
  - hsl
  - hwb
  - lab
  - lch
  - rgb
- Helpers
  - unique
- Internet
  - emoji
  - httpStatusCode
  - url
- Random
  - alpha
  - alphaNumeric
- String
  - alpha
  - alphanumeric
  - hexadecimal
````

**Review by Shinigami92 — DISMISSED** [review]

**Review by matthewmayer — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**